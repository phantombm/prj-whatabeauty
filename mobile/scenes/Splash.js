import React, { Component } from 'react';
import { View, Image, Animated } from 'react-native';
import PropTypes from 'prop-types';

export default class Splash extends Component {
  static propTypes = {
    endLoading: PropTypes.func
  };

  animatedValue = new Animated.Value(0);

  animatedOpacity = new Animated.Value(0);

  animatedTranslateY = this.animatedValue.interpolate({
    inputRange: [0, 1],
    outputRange: [30, 0]
  });

  componentDidMount() {
    setTimeout(() => {
      Animated.parallel([
        Animated.timing(
          this.animatedValue,
          {
            toValue: 1,
            duration: 500
          }
        ),
        Animated.timing(
          this.animatedOpacity,
          {
            toValue: 1,
            duration: 500
          }
        )
      ]).start();
    }, 500);
  }

  disappearSpalshImage = () => {
    Animated.timing(
      this.animatedOpacity,
      {
        toValue: 0,
        duration: 500
      }
    ).start();

    setTimeout(() => {
      this.props.endLoading();
    }, 500);
  };

  render() {
    return (
      <View style={{ flex: 1 }}>
        <View style={{
          flex: 7,
          alignItems: 'center',
          justifyContent: 'center'
        }} >
          <Animated.Image source={ require('../assets/images/splash_image.png') } style={{ opacity: this.animatedOpacity, transform: [{ translateY: this.animatedTranslateY }] }} />
        </View>
        <View style={{ flex: 3 }} />
      </View>
    );
  }
}
