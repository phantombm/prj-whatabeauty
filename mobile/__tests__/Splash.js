import React from 'react';
import 'react-native';
import testRenderer from 'react-test-renderer';

import Splash from '../scenes/Splash';

test('renders correctly', () => {
  const tree = testRenderer.create(
    <Splash />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
