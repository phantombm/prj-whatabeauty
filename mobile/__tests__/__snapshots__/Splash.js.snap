// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 7,
        "justifyContent": "center",
      }
    }
  >
    <Image
      source={1}
      style={
        Object {
          "opacity": 0,
          "transform": Array [
            Object {
              "translateY": 30,
            },
          ],
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "flex": 3,
      }
    }
  />
</View>
`;
